x-pack/platform/plugins/shared/task_manager/server/saved_objects/schemas/rrule.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { Frequency } from '@kbn/rrule';
+import moment from 'moment-timezone';
+
+function validateTimezone(timezone: string) {
+  if (moment.tz.zone(timezone) != null) {
+    return;
+  }
+
+  return 'string is not a valid timezone: ' + timezone;
+}
+
+const rruleCommon = schema.object({
+  freq: schema.number(),
+  interval: schema.number(),
+  tzid: schema.string({ validate: validateTimezone, defaultValue: 'UTC' }),
+});
+
+const byminute = schema.maybe(schema.arrayOf(schema.number({ min: 0, max: 59 })));
+const byhour = schema.maybe(schema.arrayOf(schema.number({ min: 0, max: 23 })));
+const byweekday = schema.maybe(schema.arrayOf(schema.number({ min: 1, max: 7 })));
+const bymonthday = schema.maybe(schema.arrayOf(schema.number({ min: 1, max: 31 })));
```

Comment:
The alerting schema has some validations for these fields (checking for non-empty arrays). Should we add similar validations here? Also, I know we're only supporting a subset of the rRule specification, but should we try to make it as close to the alerting one as possible? For example, here, `byweekday` is a number that maps to an enum but in the alerting spec, it's a string.

Alerting schema: `x-pack/platform/plugins/shared/alerting/common/routes/r_rule/request/schemas/v1.ts`

Reply:
cc @cnasikas Any feedback/comments on this schema?

Reply:
It would be nice if we had a schema in a package where you could import it (future work).

Reply:
OK, I used the same schemas.